src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -109,7 +112,7 @@ function _M:storeTokenInCache(cacheLookupKey, cachingObj, expire_at_ms_utc)
     ngx.log(ngx.DEBUG, "Storing a new token expiring in  " .. tostring(local_expire_in) .. " s locally, out of a total validity of " .. tostring(expires_in_s) .. " s.")
     local cachingObjString = cjson.encode(cachingObj)
     self:setKeyInLocalCache(cacheLookupKey, cachingObjString, local_expire_in, "cachedOauthTokens")
-    self:setKeyInRedis(cacheLookupKey, "token_json", expire_at_ms_utc, cachingObjString)
+    self:setKeyInRedis(cacheLookupKey, "token_json", math.min( expire_at_ms_utc - (os.time() * 1000), REDIS_CACHE_TTL ), cachingObjString)
```

Comment:
The third parameter of the `setKeyInRedis` method need to be a timestamp, not a TTL, so this should probably be ```math.min( expire_at_ms_utc, (os.time()  + MAX_REDIS_CACHE_TTL) * 1000) )``` 
(if the max TTL is specified in seconds)